rfcs/load_test.md

```diff
@@ -0,0 +1,46 @@
+# RFC 33: Crash Tests
+
+## Summary
+
+Add a test type in which the test document is simply loaded. The test
+passes if this occurs without encountering a fatal error e.g. a crash.
+
+## Details
+
+Load Tests are static tests typically used for cases where browsers
+are known to have a bug such as a crash or a leak. They differ from
+other tests such as reftests and testharness tests in that there is no
+pass condition other than the page loading successfully. In vendor
+infrastructure these tests may also be run with additional
+instrumentation e.g. address sanitizing, making the pass condition
+more stringent. The lack of an explicit pass condition makes it easier
+to write these tests than tests where a detailed understanding of the
+correct behaviour is required, making them suitable for autogeneration
+e.g. as the output of a fuzzing tool.
+
+The infrastructure changes are as follows:
+
+* A new test type `crash` recognised by the manifest and assigned to
+  any test file with a name containing `-crashtest` immediately before
+  the extension or in a subdirectory called `crashtests`.
```

Comment:
I think calling this `crashtest` goes against the pattern of `manual` and `visual` here.

The prior art here is `-crash` being used in as a filename suffix in WebKit, and `crashtests` as a directory (though combined with a `crashtests.list` manifest, rather than simply considering *everything* in the directory a test).

Reply:
`-crash` turns out to already be used a lot, possibly for things that were converted from crashtests to other types.

Note that `foo/crashtests/support/` still allows including external resources in tests without requiring an explicit manifest.

Reply:
How much is it used? If there are less than 100 then fixing/renaming them would free up this good name for use.

Reply:
```
jgraham@flitwick:~/develop/web-platform-tests$ find . -name '*-crash.*'
./2dcontext/wide-gamut-canvas/imagedata-no-color-settings-crash.html
./payment-request/payment-request-constructor-crash.https.html
./dom/nodes/remove-and-adopt-crash.html
./html/semantics/text-level-semantics/the-ruby-element/rt-without-ruby-crash.html
./html/semantics/interactive-elements/the-details-element/display-table-with-rt-crash.html
./html/semantics/interactive-elements/the-summary-element/display-table-with-rt-crash.html
./html/semantics/forms/the-input-element/time-datalist-crash.html
./html/semantics/forms/the-input-element/type-change-file-to-text-crash.html
./html/semantics/embedded-content/the-object-element/block-object-with-ruby-crash.html
./editing/run/first-letter-crossing-engine-boundary-crash.html
./svg/svg-in-svg/svg-in-svg-circular-filter-reference-crash.html
./svg/extensibility/foreignObject/foreign-object-under-defs-crash.html
./svg/extensibility/foreignObject/foreign-object-circular-filter-reference-crash.html
./svg/extensibility/foreignObject/foreign-object-under-clip-path-crash.html
./svg/pservers/pattern-with-invalid-base-cloned-crash.html
./css/css-pseudo/first-letter-of-html-root-crash.html
./css/css-pseudo/before-in-display-none-crash.html
./css/css-pseudo/first-letter-crash.html
./css/css-pseudo/first-line-first-letter-insert-crash.html
./css/css-animations/keyframes-remove-documentElement-crash.html
./css/css-position/position-absolute-in-inline-crash.html
./css/css-images/infinite-radial-gradient-crash.html
./css/css-images/gradient-crash.html
./css/css-tables/visibility-collapse-rowspan-crash.html
./css/css-tables/visibility-collapse-colspan-crash.html
./css/css-typed-om/set-var-reference-crash.html
./css/css-writing-modes/bidi-inline-fragment-crash.html
./css/css-overflow/shrink-to-fit-auto-overflow-relayout-crash.html
./css/css-overflow/outline-with-opacity-crash.html
./css/css-shapes/shape-outside/supported-shapes/inset/shape-outside-inset-crash.html
./css/css-shapes/shape-outside/supported-shapes/polygon/shape-outside-polygon-crash.html
./css/CSS2/floats-clear/adjoining-float-new-fc-crash.html
./css/CSS2/floats/floats-saturated-position-crash.html
./css/CSS2/linebox/video-needs-layout-crash.html
./css/css-multicol/with-custom-layout-on-same-element-crash.https.html
./css/css-multicol/extremely-tall-multicol-with-extremely-tall-child-crash.html
./css/css-multicol/abspos-in-multicol-with-spanner-crash.html
./css/css-break/break-before-with-no-fragmentation-crash.html
./css/css-text/overflow-wrap/overflow-wrap-break-word-long-crash.html
./css/css-text/overflow-wrap/overflow-wrap-break-word-white-space-crash.html
./css/css-text/white-space/nowrap-wbr-and-space-crash.html
./css/css-text/white-space/pre-line-br-with-whitespace-child-crash.html
./css/css-text/white-space/pre-with-whitespace-crash.html
./css/css-text/text-indent/text-indent-long-line-crash.html
./quirks/table-replaced-descendant-percentage-height-crash.html
./portals/portals-no-frame-crash.html
```

Reply:
So:

We currently have 46 `'*-crash.*'` files. 42 of these include testharness.js, the other 4 are reftests.

Roughly half of the testharness.js tests are already `test(()=>{})` tests. Only 4 tests actually call any assert function.

As such, that's only eight tests that need renamed.